src/glob.zig

```diff
@@ -0,0 +1,3032 @@
+        _ = pattern;
+        // Collapse successive double wildcards
+        while (component_idx + 1 < this.patternComponents.items.len and
+            this.patternComponents.items[component_idx + 1].syntax_hint == .Double) : (component_idx += 1)
+        {}
+        return component_idx;
+    }
+
+    pub fn skipSpecialComponents(
+        this: *GlobWalker,
+        work_item_idx: u32,
+        dir_path: *[:0]u8,
+        scratch_path_buf: *[bun.MAX_PATH_BYTES]u8,
+        encountered_dot_dot: *bool,
+    ) u32 {
+        var component_idx = work_item_idx;
+
+        // Skip `.` and `..` while also appending them to `dir_path`
+        component_idx = switch (this.patternComponents.items[component_idx].syntax_hint) {
+            .Dot => this.collapseDots(
+                component_idx,
+                dir_path,
+                scratch_path_buf,
+                encountered_dot_dot,
+            ),
+            .DotBack => this.collapseDots(
+                component_idx,
+                dir_path,
+                scratch_path_buf,
+                encountered_dot_dot,
+            ),
+            else => component_idx,
+        };
+
+        // Skip to the last `**` if there is a chain of them
+        component_idx = switch (this.patternComponents.items[component_idx].syntax_hint) {
+            .Double => this.collapseSuccessiveDoubleWildcards(component_idx),
+            else => component_idx,
+        };
+
+        return component_idx;
+    }
+
+    pub fn handleDirEntriesRoot(
+        this: *GlobWalker,
+        work_item: *const WorkItem,
+        scratch_path_buf: *[bun.MAX_PATH_BYTES]u8,
+        cwd_fd: bun.FileDescriptor,
+    ) !Maybe(u0) {
+        scratch_path_buf[0] = 0;
+        var dir_path: [:0]u8 = dir_path: {
+            // For the root `work_item.path` is absolute, so if absolute is
+            // disabled, make this an empty string so we joining dir with
+            // entries isn't absolute
+            if (!this.absolute) break :dir_path scratch_path_buf[0..0 :0];
+
+            @memcpy(scratch_path_buf[0..work_item.path.len], work_item.path);
+            scratch_path_buf[work_item.path.len] = 0;
+            break :dir_path scratch_path_buf[0..work_item.path.len :0];
+        };
+
+        var encountered_dot_dot = false;
+        const component_idx = this.skipSpecialComponents(work_item.idx, &dir_path, scratch_path_buf, &encountered_dot_dot);
+
+        // If we encountered a `..` then we need to change the fd
+        if (encountered_dot_dot) {
+            const flags = std.os.O.DIRECTORY | std.os.O.RDONLY;
+            const fd = switch (Syscall.openat(cwd_fd, dir_path, flags, 0)) {
+                .err => |err| return .{
+                    .err = this.handleSysErrWithPath(err, dir_path),
+                },
+                .result => |fd_| fd_,
+            };
+            defer {
+                _ = Syscall.closeAllowingStdoutAndStderr(fd);
```

Comment:
This can just be `close`, it should never be stdout or stderr. If it is, that is a bug.
```suggestion
                _ = Syscall.close(fd);
```